src/store/workflows.module.js

```diff
@@ -40,6 +42,28 @@ function _getCycles (workflows) {
   return cycles
 }
 
+/**
+ * Compute percent progress.
+ *
+ * @see https://github.com/cylc/cylc-flow/blob/de7d938496e82dbdfb165938145670dd8e801efd/lib/cylc/gui/updater_tree.py#L248-L263
+ * @param {number} startedTime in milliseconds since 1970-01-01 00:00:00 UTC, e.g. 1568353099874
+ * @param {number} meanElapsedTime mean elapsed time in seconds
+ * @returns {number} the percent progress, e.g. 25 (meaning 25% progress)
+ * @private
+ */
+function _computePercentProgress (startedTime, meanElapsedTime) {
+  const now = Date.now() // milliseconds since 1970-01-01
```

Comment:
How does time zone affect this?

Reply:
Oh, good point!!

I think we have a bug! The new Date here will use client TZ. While the started time I think could be in any TZ?

Reply:
The started time string always has a time zone. Does `Date.parse` recognise and handle that automatically? If so, the subsequent duration calculation should be correct regardless?

Reply:
Oh, could be! I've added your comment here https://github.com/cylc/cylc-ui/issues/237 :+1: hopefully that issue can be closed as not-an-issue :grimacing:

Reply:
Discussion moved to #237 ...